packages/@aws-cdk/aws-redshift/lib/table.ts

```diff
@@ -0,0 +1,177 @@
+import * as cdk from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { ICluster } from './cluster';
+import { DatabaseProps } from './database-props';
+import { DatabaseQuery } from './private/database-query';
+import { TableHandlerProps } from './private/handler-props';
+import { TableAction } from './privileges';
+import { IUser } from './user';
+
+// keep this import separate from other imports to reduce chance for merge conflicts with v2-main
+// eslint-disable-next-line no-duplicate-imports, import/order
+import { Construct as CoreConstruct } from '@aws-cdk/core';
+
+/**
+ * A column in a Redshift table.
+ */
+export interface Column {
+  /**
+   * The name of the column.
+   */
+  readonly name: string;
+
+  /**
+   * The data type of the column.
+   */
+  readonly dataType: string;
+}
+
+/**
+ * Properties for configuring a Redshift table.
+ */
+export interface TableProps extends DatabaseProps {
+  /**
+   * The name of the table.
+   *
+   * @default - a name is generated
+   */
+  readonly tableName?: string;
+
+  /**
+   * The columns of the table.
+   */
+  readonly tableColumns: Column[];
+
+  /**
+   * The policy to apply when this resource is removed from the application.
+   *
+   * @default cdk.RemovalPolicy.Retain
+   */
+  readonly removalPolicy?: cdk.RemovalPolicy;
+}
+
+/**
+ * Represents a table in a Redshift database.
+ */
+export interface ITable extends cdk.IConstruct {
+  /**
+   * Name of the table.
+   */
+  readonly tableName: string;
+
+  /**
+   * The columns of the table.
+   */
+  readonly tableColumns: Column[];
+
+  /**
+   * The cluster where the table is located.
+   */
+  readonly cluster: ICluster;
+
+  /**
+   * The name of the database where the table is located.
+   */
+  readonly databaseName: string;
+
+  /**
+   * Grant a user privilege to access this table.
+   */
+  grant(user: IUser, ...actions: TableAction[]): void;
+}
+
+/**
+ * A full specification of a Redshift table that can be used to import it fluently into the CDK application.
+ */
+export interface TableAttributes {
+  /**
+   * Name of the table.
+   */
+  readonly tableName: string;
+
+  /**
+   * The columns of the table.
+   */
+  readonly tableColumns: Column[];
+
+  /**
+   * The cluster where the table is located.
+   */
+  readonly cluster: ICluster;
+
+  /**
+   * The name of the database where the table is located.
+   */
+  readonly databaseName: string;
+}
+
+abstract class TableBase extends CoreConstruct implements ITable {
```

Comment:
Make this class internal. No need to expose it in the public API.

Same for other base classes.

Reply:
It's not `export`d so shouldn't be exposed?

Reply:
You're right. I missed that.